### PR TITLE
Accepts non zero check done in internal-calls.

### PR DESCRIPTION
**Pull request by feliam**
Detects (or silence the detector) when the zero address check is done inside an internal call.
This also take into account some data dependency. 
Unittests for the detector were updated. 
[ Fixes 1030](https://github.com/crytic/slither/issues/1030)

### CODE REVIEW

File: slither/detectors/operations/missing_zero_address_validation.py [hunk 1]
```diff
@@ -72,17 +72,37 @@ def _zero_address_validation(self, var, node, explored):
...
                 return True
         return False
 
+    def _zero_address_validation_icalls(self, var, node) -> bool:
+        """ Check if the interesting variable `var` is checked inside any `Call` expression
+            in this `node`
+        """
+        if node.type == node.type.EXPRESSION:
+            for ir in node.irs:
+                if isinstance(ir, Call):
```
Comment (feliam) [inline slither/detectors/operations/missing_zero_address_validation.py:96]: Call, InternalCall or libraryCall  ??

**Review by fasya10 — APPROVED** [review]

File: slither/detectors/operations/missing_zero_address_validation.py [hunk 2]
```diff
@@ -72,17 +72,36 @@ def _zero_address_validation(self, var, node, explored):
...
         for father in node.fathers:
             if self._zero_address_validation(var, father, explored):
                 return True
         return False
 
+    def _zero_address_validation_icalls(self, var, node) -> bool:
+        """Check if the interesting variable `var` is checked inside any `Call` expression
+        in this `node`
+        """
+        if node.type == node.type.EXPRESSION:
```
Comment (0xalpharush) [inline slither/detectors/operations/missing_zero_address_validation.py:93]: What is this supposed to do? It seems unnecessary

File: slither/detectors/operations/missing_zero_address_validation.py [hunk 3]
```diff
@@ -72,17 +72,36 @@ def _zero_address_validation(self, var, node, explored):
...
+            for ir in node.irs:
+                if isinstance(ir, Call):
```
Comment (0xalpharush) [inline slither/detectors/operations/missing_zero_address_validation.py:95]: Can this be `InternalCall`?

File: slither/detectors/operations/missing_zero_address_validation.py [hunk 4]
```diff
@@ -72,17 +72,36 @@ def _zero_address_validation(self, var, node, explored):
...
-            var in node.variables_read
-        ):
-            return True
+        if node.contains_if():
+            if var in node.variables_read:
+                return True
+            # check dependant variables
+        if node.contains_require_or_assert():
+            if var in node.variables_read:
+                return True
```
Comment (0xalpharush) [inline slither/detectors/operations/missing_zero_address_validation.py:81]:
```suggestion
        if node.contains_if() or node.contains_require_or_assert():
            if var in node.variables_read:
                return True
```

**Pull request closed without merging**